app/services/update_transaction.rb

```diff
@@ -0,0 +1,27 @@
+class UpdateTransaction
+  attr_accessor :transaction
+
+  def initialize(transaction:)
+    self.transaction = transaction
+  end
+
+  def call(attributes: {})
+    transaction.assign_attributes(attributes)
+    transaction.value = sanitize_monetary_string(value: attributes[:value])
+
+    if transaction.valid?
+      transaction.save
```

Comment:
I often wonder about `save` like this. What happens if for some reason the save fails, do we just trust it works?

Reply:
This is a fair concern.

In thinking about it more I wonder if we should change it around a bit to:

```
result = Result.new(transaction.save, transaction)
```

Since `transaction.save` _can_ return false theoretically. The other way might be to `transaction.save!` and terminate the execution chain and fail loudly. What do you think?

Reply:
I've made sure if transaction.save returns false unexpectedly, we return a falsey Result.

Reply:
To be honest, we trust `save` all over the place - I was interested in what your take was. I think because `save` calls `valid?` I get hung up on the duplication, but I like the way the code feels with the explicit call to `vaid?` – it tells the story nicely.

I don't think we should worry here - this was a great piece of work - maybe chat about `save` over a meyricano™ sometime...